src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -22,16 +24,24 @@ class LeonardoService(gdDAO: DataprocDAO, dbRef: DbReference)(implicit val execu
   }
 
   def createCluster(googleProject: String, clusterName: String, clusterRequest: ClusterRequest): Future[Cluster] = {
-    gdDAO.createCluster(googleProject, clusterName, clusterRequest) flatMap { clusterResponse: ClusterResponse =>
+    def create = gdDAO.createCluster(googleProject, clusterName, clusterRequest) flatMap { clusterResponse: ClusterResponse =>
       dbRef.inTransaction { dataAccess =>
         dataAccess.clusterQuery.save(Cluster(clusterRequest, clusterResponse))
       }
     }
+
+    dbRef.inTransaction { dataAccess =>
+      dataAccess.clusterQuery.getByName(googleProject, clusterName)} flatMap {
+      case Some(c: Cluster) => throw new ClusterNotFoundException(googleProject, clusterName)
```

Comment:
It seems weird to throw a `ClusterNotFoundException` when a Some is returned. Could we make a new exception like `ClusterAlreadyExistsException`?

Reply:
agreed. this is literally a wrapper around 409 conflict